rskj-core/src/main/java/org/ethereum/rpc/exception/RskErrorResolver.java

```diff
@@ -21,16 +22,32 @@
 
     @Override
     public JsonError resolveError(Throwable t, Method method, List<JsonNode> arguments) {
-        JsonError error = null;
-        if(t instanceof  RskJsonRpcRequestException) {
-            error =  new JsonError(((RskJsonRpcRequestException) t).getCode(), t.getMessage(), null);
+        JsonError error;
+
+        if (t instanceof RskJsonRpcRequestException) {
+            error = new JsonError(((RskJsonRpcRequestException) t).getCode(), t.getMessage(), null);
         } else if (t instanceof InvalidFormatException) {
             error = new JsonError(JsonRpcError.INTERNAL_ERROR, "Internal server error, probably due to invalid parameter type", null);
         } else if (t instanceof UnrecognizedPropertyException) {
             error = new JsonError(
                     JsonRpcError.INVALID_PARAMS,
                     getExceptionMessage((UnrecognizedPropertyException) t),
                     null);
+        } else if (t.getMessage() != null && t.getMessage().contains("An RSK address must be")) {
+            error = new JsonError(
+                    JsonRpcError.INVALID_PARAMS,
+                    "invalid argument 0: hex string has length " + arguments.get(0).asText().replace("0x", "").length() + ", want 40 for common.Address",
```

Comment:
not sure we should return messages to a user with implementation details, e.g. `common.Address`

Reply:
That's the message geth returns, should I updated?

Reply:
yeah, I believe we should not return names of structs defined in the geth source code

Reply:
Addressed